src/js/control-bar/picture-in-picture-toggle.js

```diff
@@ -56,7 +43,30 @@ class PictureInPictureToggle extends Button {
    *         The DOM `className` for this object.
    */
   buildCSSClass() {
-    return `vjs-picture-in-picture-control ${super.buildCSSClass()}`;
+    return `vjs-picture-in-picture-control vjs-hidden ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * Displays or hides the button depending on the audio mode detection.
+   * Exits picture-in-picture if it is enabled when switching to audio mode.
+   */
+  handlePictureInPictureAudioModeChange() {
+    // This audio detection will not detect HLS or DASH audio-only streams because there was no reliable way to detect them at the time
+    const isSourceAudio = this.player_.currentType().substring(0, 5) === 'audio';
```

Comment:
We should probably update `isAudio_` when this matches as the source is set, then use `isAudio()` here.

Reply:
@mister-ben it make sens. Your suggestion is to set the audio flag somewhere in 
https://github.com/videojs/video.js/blob/f1558c660bbc132b23dbd63b2f3cdf7b6e60d00b/src/js/player.js#L1582-L1629

Do you recommend doing it in another PR? If yes with what kind of scope? feat, refactor or fix?

I would tend to create another PR to keep modification more atomic. This would allow to try to better handle the HLS and Dash audio.